ci: add luacheck linter

themilchenko opened the pull request:
There was a linter check in test in `testing` workflow, but not in a separate job.

After the patch a linter check is available in a separate job and the previous step from `testing` workflow was deleted.

Closes #21

themilchenko committed:
ci: add luacheck linter

There was a linter check in test in `testing` workflow, but not in a
separate job.

After the patch a linter check is available in a separate job and the
previous step from `testing` workflow was deleted.

Closes #21

DifferentialOrange commented on .github/workflows/check.yaml:
```diff
@@ -0,0 +1,37 @@
+
+    - run: echo $PWD/.rocks/bin >> $GITHUB_PATH
+
+    - name: Run luacheck
```
I think we should run full `make check` here, since we removed `make check` from the other workflow.

themilchenko replied:
Ok, but `check` is an alias for `luacheck`:

https://github.com/tarantool/metrics-export-role/blob/d31d99a4e5d84b9de9d386374dface43bc75ae1e/Makefile#L33-L36

DifferentialOrange replied:
For now it is, but it is possible that later it would be something like
```make
check: luacheck codespell
```

themilchenko replied:
Yes, corrected.

DifferentialOrange commented on .github/workflows/check.yaml:
```diff
@@ -0,0 +1,37 @@
+
+    - name: Setup luacheck
+      run: |
+        tt rocks install luacheck 0.26.0
```
If would be nice to separate Makefile lint deps and test deps. But it's up to you. You also can see this example: https://github.com/tarantool/sideservice/blob/master/Makefile

themilchenko replied:
Decompose an installation of deps in a separate commit.

themilchenko committed:
makefile: decompose deps installation

Since it will be more convenient to see CI jobs making its own work, the
installation of rocks was rewriten to install rocks only for one task.

oleg-jukovec commented on Makefile:
```diff
@@ -51,9 +51,24 @@ coveralls: $(LUACOV_STATS)
 	tt rocks install luacov 0.13.0-1
 	tt rocks install luacov-coveralls 0.2.3-1 --server=http://luarocks.org
+
+deps-lint:
+	tt rocks install luacheck 0.26.0
+
+deps:
+ifeq ($(depname), test)
+	$(MAKE) deps-test
+else ifeq ($(depname), coverage)
+	$(MAKE) deps-coverage
+else ifeq ($(depname), lint)
+	$(MAKE) deps-lint
+else
+	$(MAKE) deps-coverage deps-lint
+endif
 	tt rocks make
```
Please, update the `Development` article in the README.md.

themilchenko replied:
Updated.

DifferentialOrange commented on .github/workflows/check.yaml:
```diff
@@ -0,0 +1,36 @@
+
+    - run: echo $PWD/.rocks/bin >> $GITHUB_PATH
+
+    - name: Run luacheck
```
Now it's "Run static analysis"

themilchenko replied:
Edited.

DifferentialOrange commented on README.md:
````diff
@@ -206,7 +206,28 @@ git clone https://github.com/tarantool/metrics-export-role
+
+```
+
+Finally to install all of this, simply run:
````
It's better to start with this one

themilchenko replied:
Edited.

DifferentialOrange commented on README.md:
````diff
@@ -206,12 +206,32 @@ git clone https://github.com/tarantool/metrics-export-role
 ```
 
-After that you need to install dependencies (`tt` is required):
+After that you need to install dependencies (`tt` is required).
+It's possible to install requirements only for running tests
+or linter check. To install all of this, run:
 
 ```shell
 make deps
 ```
 
+To install only tests requirements, run:
+
+```shell
+make deps depname=test
+```
+
+To install tests with coverage requirements, run:
````
````suggestion
After that you need to install dependencies (`tt` is required).
To do it, run:

```shell
make deps
```
It's possible to install requirements only for running tests
or linter check. To install only tests requirements, run:

```shell
make deps depname=test
```

To install tests with coverage requirements, run:
````

themilchenko replied:
Edited.
